index.js

```diff
@@ -43,11 +43,19 @@ function repeat(str, num) {
     throw new TypeError('expected a string');
   }
 
+  // no need to repeat an empty string
+  if (str === '') return '';
+
   // cover common, quick use cases
   if (num === 1) return str;
   if (num === 2) return str + str;
 
   var max = str.length * num;
+
+  if (max === Infinity) {
```

Comment:
Nice work. Please move this check to be just after the string check (line 46 or so) and change it to `if (num === Infinity)`.

Reply:
The reason I put it on that line was to leverage the implicit "parseFloat()" when `str.length` and `num` are multiplied together e.g.

```javascript
5 * Infinity;
//=> Infinity

5 * 'Infinity';
//=> Infinity

5 * '    Infinity   ';
//=> Infinity
```

That was also motivated by some of the tests that seem to suggest that `num` can be given as a string too. I'm happy to make the change but should I still care about the case of `Infinity` as a string then?

By the way it seems that if `num` can be given as a string then the two shortcuts at L50,51 are most likely ignored. Should this be addressed?

I am happy to make the change you're suggesting but should I also test for string case as `num` can be given as a string too (at least from what I can see in the docs)